### PR TITLE
Add functions to delete all messages from the feed.

### PR DESCRIPTION
**Pull request by saschadoemer**
Format.

### CODE REVIEW

**Review by Frank-Wiebeler — CHANGES_REQUESTED** [review]
Please fix CI first

File: agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/http/DeleteMessageService.java [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+import com.dke.data.agrirouter.api.dto.onboard.OnboardingResponse;
+import com.dke.data.agrirouter.api.messaging.HttpAsyncMessageSendingResult;
 import com.dke.data.agrirouter.api.service.parameters.DeleteMessageParameters;
 
-public interface DeleteMessageService extends MessagingService<DeleteMessageParameters> {}
+public interface DeleteMessageService extends MessagingService<DeleteMessageParameters> {
+
+  /**
+   * Delete all messages in the feed. The function will delete all messages based on the time
+   * intervall of 4 weeks (which is the maximum of time the messages are stored within the
```
Comment (Frank-Wiebeler) [inline agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/http/DeleteMessageService.java:11]: -l (interval)

File: agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/http/DeleteMessageService.java [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+   * agrirouter).
+   *
+   * @param onboardingResponse The onboard response for the endpoint.
+   * @return The message ID.
+   */
+  String deleteAll(OnboardingResponse onboardingResponse);
+
+  /**
+   * Delete all messages in the feed. The function will delete all messages based on the time
+   * intervall of 4 weeks (which is the maximum of time the messages are stored within the
```
Comment (Frank-Wiebeler) [inline agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/http/DeleteMessageService.java:21]: -l ;)

File: agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/mqtt/DeleteMessageService.java [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+import com.dke.data.agrirouter.api.dto.onboard.OnboardingResponse;
+import com.dke.data.agrirouter.api.messaging.MqttAsyncMessageSendingResult;
 import com.dke.data.agrirouter.api.service.parameters.DeleteMessageParameters;
 
-public interface DeleteMessageService extends MessagingService<DeleteMessageParameters> {}
+public interface DeleteMessageService extends MessagingService<DeleteMessageParameters> {
+
+  /**
+   * Delete all messages in the feed. The function will delete all messages based on the time
+   * intervall of 4 weeks (which is the maximum of time the messages are stored within the
```
Comment (Frank-Wiebeler) [inline agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/mqtt/DeleteMessageService.java:11]: -l ;)

File: agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/mqtt/DeleteMessageService.java [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+   * agrirouter).
+   *
+   * @param onboardingResponse The onboard response for the endpoint.
+   * @return The message ID.
+   */
+  String deleteAll(OnboardingResponse onboardingResponse);
+
+  /**
+   * Delete all messages in the feed. The function will delete all messages based on the time
+   * intervall of 4 weeks (which is the maximum of time the messages are stored within the
```
Comment (Frank-Wiebeler) [inline agrirouter-sdk-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/mqtt/DeleteMessageService.java:21]: -l ;)

**Review by Frank-Wiebeler — COMMENTED** [review]
Just small typos; can be merged after those were fixed.

DRY principle is violated ( createMessageParametersToDeleteAllMessages ); this should maybe be addressed in an additional PR as it is the case with the other 2 PRs as well.


In general:   Awesome work, thank you for your input ヽ(´▽`)/

**Comment by saschadoemer** [comment]
> Just small typos; can be merged after those were fixed.
> 
> DRY principle is violated ( createMessageParametersToDeleteAllMessages ); this should maybe be addressed in an additional PR as it is the case with the other 2 PRs as well.
> 
> In general: Awesome work, thank you for your input ヽ(´▽`)/

Fixed the DRY problem with a default implementation on the interface. Not quite sure about the abstraction overhead.

**Review by Frank-Wiebeler — APPROVED** [review]
Awesome as always 🥳

**Pull request merged**